source/opt/constants.h

```diff
@@ -325,22 +325,39 @@ class ConstantManager {
   // points to the same instruction before and after the insertion. This is the
   // only method that actually manages id creation/assignment and instruction
   // creation/insertion for a new Constant instance.
+  //
+  // |type_id| is an optional argument for disambiguating equivalent types. If
+  // |type_id| is specified, it is used as the type of the constant. Otherwise
+  // the type of the constant is derived by getting an id from the type manager
+  // for |c|.
```

Comment:
I needed to do this because the TypeManager doesn't have a reliable id mapping for ambiguous types. Users of the constant manager do. This is a drawback to be able to search for id-less types.